src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.tsx

```diff
@@ -28,7 +33,9 @@ export const ServiceInstanceForm: React.FC<Props> = ({
       ? createEditFormState(fields, originalAttributes)
       : createFormState(fields)
   );
-
+  const [dirtyInputs, setDirtyInputs] = useState<string[]>([]);
```

Comment:
Why the array? 
The getUpdate is being called on any change in any input field, so you can by default assume the isDirty state can be set to True.

Reply:
Yeah, but we should be able to tell if user clear all data manually, Uncommon but not impossible

Reply:
if we would base dirty state then I don't see easy way to tell when to change it back to false

Reply:
yeah, but performance wise this is like a huge difference... imagine having an instance with a really complex form with a lot of fields and nested fields... You'll be comparing continuously the difference on each change... Each letter that you write is triggering the getUpdate.

If you keep it simple and set isDirty on the first change you catch, you only need to set it once, and can make a check If IsDirty just continue executing rest of the code, otherwise set it to true.

The case where the user will be clearing the whole form manually is 1/XXXXXX They'd rather click the cancel button in that case.

Reply:
It will, cause currently your logic is being run on each input field that is being touched. You set a state which causes rerenders on each input change. 

While on the other hand if you only have 1 boolean attribute that is being set once, and causes 1 rerender.